sdk/core/azure-core/src/io/body_stream.cpp

```diff
@@ -190,3 +190,32 @@ size_t FileBodyStream::OnRead(uint8_t* buffer, size_t count, Azure::Core::Contex
 void FileBodyStream::Rewind() { m_randomAccessFileBodyStream->Rewind(); }
 
 int64_t FileBodyStream::Length() const { return m_randomAccessFileBodyStream->Length(); }
+
+ProgressBodyStream::ProgressBodyStream(
+    BodyStream& bodyStream,
+    std::function<void(int64_t bytesTransferred)> callback)
+    : m_bodyStream(bodyStream), m_bytesTransferred(0), m_callback(std::move(callback))
+{
+  m_callback(m_bytesTransferred);
```

Comment:
Are we sure we should call the callback on construction when no call to `OnRead()` has happened yet?